rust_src/src/keymap.rs

```diff
@@ -50,6 +52,33 @@ pub fn keymapp(object: LispObject) -> bool {
     map.is_not_nil()
 }
 
+/// Return the parent map of KEYMAP, or nil if it has none.
+/// We assume that KEYMAP is a valid keymap.
+#[no_mangle]
+pub extern "C" fn keymap_parent(keymap: Lisp_Object, autoload: bool) -> Lisp_Object {
+    let map = keymap_parent_internal(LispObject::from_raw(keymap), autoload);
+    map.to_raw()
+}
+
+fn keymap_parent_internal(keymap: LispObject, autoload: bool) -> LispObject {
+    let mut list = LispObject::constant_nil();
+    let map = unsafe { LispObject::from_raw(get_keymap(keymap.to_raw(), true, autoload)) };
+    for elt in map.iter_tails_safe() {
+        list = elt.cdr();
+        if keymapp(list) {
+            return list;
+        }
+    }
```

Comment:
Could the `find` method of `Iterator` be used here instead?

Reply:
Yes. And it seems that it would even be more concise. Will be refactoring this.